### DISCUSSION TITLE
RFC : SCORM Support in Sunbird ED

### DISCUSSION THREAD

**Discussion by parthlawate** [Sunbird-Ed/Community#309] [Feature request]
Sunbird currently does not support SCORM format. This format is currently the most widely adopted format of E-learning content development .  There are industry standard content development tools like Articulate Storyline, Adobe Captivate, ISpring etc that elearning content developers are very conversant with. A lot of content is ready in this format with organisations. 

Its therefore important that Sunbird support this format with detailed tracking so adopters can leverage existing content & existing skills sets & tools for developing content. 

I see the following areas where changes would be required 

a) Changes to editor to allow for uploading SCORM packages
b) Changes to content service publish pipeline to process SCORM format content
c) Changes to player to play SCORM content
d) Necessary instrumentation changes to allow for tracking SCORM progress so that the same can be reported in progress exhausts and for users to resume from where they have left in the previous session

Some related Topics posted earlier on this for reference : 
https://github.com/project-sunbird/sunbird-community/discussions/701
https://github.com/project-sunbird/sunbird-community/discussions/628

Note that a SCORM Implementation  was done in Sunbird CB but from our initial calls with those teams, it needs the SCORM to be created in a certain authoring tool & published in a certain way .. so its not universal. However we could use it as a starter if the code is available post review.

**Comment by Krishnaj20** [comment]
thanks @parthlawate for starting this thread. 
@vaivk369 - thanks for volunteering to work on the tech design. If you want to understand more about the SunbirdCB contribution please reach out to @HarishGangula. Harish has not worked on the SunbirdCB contribution but he knows what was done by the SunbirdCB team. 

@vinukumar-vs

**Comment by vaivk369** [comment]
Hi all,
I will connect with Harish for further details.

**Comment by Krishnaj20** [comment]
@vaivk369 - let us know when you are targeting to present the tech design. Thanks

**Comment by vinukumar-vs** [comment]
@parthlawate @Krishnaj20 This has prioritised for the current release of CoKreat BB. 
@vaivk369 is working on the design from Sunbird point of view to support SCORM contents.

@luckynvdu

**Comment by vaivk369** [comment]
@Krishnaj20 https://project-sunbird.atlassian.net/wiki/spaces/SBDES/pages/3267330060/Enable+player+to+support+scorm this is the design we have started with. I am still working on this. 
@vinukumar-vs @parthlawate

**Reply by parthlawate** [reply]
adding @Manojlondhe
